src/chain.jl

```diff
@@ -0,0 +1,57 @@
+# Hanging Chain
+
+# Find the chain (of uniform density) of length L suspended between two points with minimal
+# potential energy.
+
+#   This is problem 4 in the COPS (Version 3) collection of 
+#   E. Dolan and J. More'
+#   see "Benchmarking Optimization Software with COPS"
+#   Argonne National Labs Technical Report ANL/MCS-246 (2004)
+
+function chain_model(args...; n::Int = default_nvar, kwargs...)
+  nh = max(2, div(n - 4, 4))
+
+  L = 4
+  a = 1
+  b = 3
+  tmin = b > a ? 1 / 4 : 3 / 4
+  tf = 1.0
+  h = tf / nh
+
+  nlp = Model()
+
+  @variable(nlp, u[k = 1:(nh + 1)], start = 4 * abs(b - a) * (k / nh - tmin))
+  @variable(nlp, x1[k = 1:(nh + 1)], start = 4 * abs(b - a) * k / nh * (1 / 2 * k / nh - tmin) + a)
+  @variable(
+    nlp,
+    x2[k = 1:(nh + 1)],
+    start =
+      (4 * abs(b - a) * k / nh * (1 / 2 * k / nh - tmin) + a) * (4 * abs(b - a) * (k / nh - tmin))
+  )
+  @variable(nlp, x3[k = 1:(nh + 1)], start = 4 * abs(b - a) * (k / nh - tmin))
+
+  @NLobjective(nlp, Min, x2[nh + 1])
```

Comment:
We have to formulate the problem with `ScalarNonlinearFunction` in MOI to support ExaModels. As such, we have to formulate all nonlinear expressions with `@objective` and `@constraint`.